### PR TITLE
[Rust Server] Bugfix #5947 multi tag generate multi method issue.

### PR DESCRIPTION
**Pull request by paladinzh**
#5947  [BUG] [Rust Server] Generated code with yaml multi tag compile error.

### Rust Server Technical Committee

- @frol
- @farcaller 
- @bjgill

### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [x] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.3.x`, `5.0.x`. Default: `master`.
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### CODE REVIEW

**Comment by paladinzh** [comment]
#5947 fix

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RustServerCodegen.java [hunk 1]
```diff
@@ -1056,6 +1056,26 @@ public boolean isDataTypeFile(final String dataType) {
...
+     * @param operation    OAS Operation object
+     * @param co           Codegen Operation object
+     * @param operations   map of Codegen operations
+     */
+    @SuppressWarnings("static-method")
+    public void addOperationToGroup(String tag, String resourcePath, Operation operation, CodegenOperation
+            co, Map<String, List<CodegenOperation>> operations) {
+        // only generate operation for the first tag of the tags
+        if (tag != null && co.tags.size() > 1 && !tag.equals(co.tags.get(0).getName())) {
+            LOGGER.warn("generated skip operationId `" + co.operationId + "` with the repeating tag=" + tag);
```
Comment (richardwhiuk) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RustServerCodegen.java:1073]: I don't think this is a warning - this is a perfectly fine situation to exist.

Also, I think we should say `additional`, not `repeating`.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RustServerCodegen.java [hunk 2]
```diff
@@ -1056,6 +1056,26 @@ public boolean isDataTypeFile(final String dataType) {
...
+    /**
+     * Add operation to group
+     *
+     * @param tag          name of the tag
+     * @param resourcePath path of the resource
+     * @param operation    OAS Operation object
+     * @param co           Codegen Operation object
+     * @param operations   map of Codegen operations
+     */
+    @SuppressWarnings("static-method")
```
Comment (richardwhiuk) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RustServerCodegen.java:1068]: Can you add an `@Override` annotation here? That way if the method we are overloading changes, we'll get a compile error.

Comment (paladinzh) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RustServerCodegen.java:1068]: ok. I will add it.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RustServerCodegen.java (hunk 1 above)
Comment (paladinzh) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RustServerCodegen.java]: LOGGER.info("generated skip additional tag `" + tag + "` with operationId=" + co.operationId);

**Pull request merged**